langlib/langlib-test/src/test/resources/test-src/type-param/type_param_test_negative.bal

```diff
@@ -137,3 +137,35 @@ function testStringIntFloatSimpleAndArrayUnionReturnParameterNarrowing() {
         [int, (int|float)][] y = arr.enumerate();
     }
 }
+
+type DataRecord record {|
+    string name;
+|};
+
+type DataRow record {|
+    string name;
+|};
+
+function foo(table<DataRecord> a) returns table<DataRow> {
+    table<DataRow> aa = a.map(function(dRecord) returns DataRow {
```

Comment:
Shall we add similar tests for langlib functions with one typeparam as well?

Reply:
Added more tests